pandas/tests/io/test_common.py

```diff
@@ -411,3 +414,151 @@ def test_is_fsspec_url():
     assert not icom.is_fsspec_url("random:pandas/somethingelse.com")
     assert not icom.is_fsspec_url("/local/path")
     assert not icom.is_fsspec_url("relative/local/path")
+
+
+def test_plain_text_read_csv_http_custom_headers():
```

Comment:
it is probably good to add a testcase for `read/to_parquet` as well as they have a slightly different IO handling (using `get_handle` only if needed).